File: src/main/java/cz/linkedlist/ContinuousCheckingService.java
package cz.linkedlist;

import org.springframework.scheduling.Trigger;

import java.util.Collection;
import java.util.List;

/**
 * @author Martin Macko <https://github.com/LinkedList>.
 */
public interface ContinuousCheckingService {

    /**
     * Method for registering new continuous checking task of desired tile.
     * @param utm utm of the tile
     * @param cloudiness desired minimal cloudiness in percents
     * @param contents
     */
    void register(final UTMCode utm, final Double cloudiness, List<TileSet.Contents> contents);

    /**
     * Method for registering new continuous checking task of desired tile.
     * @param utm utm of the tile
     * @param cloudiness desired minimal cloudiness in percents
     * @param contents
     * @param cron trigger expression
     */
    void register(final UTMCode utm, final Double cloudiness, List<TileSet.Contents> contents, Trigger cron);

    /**
     * Lists all currently registered tasks
     * @return list of all registered tasks
     */
    Collection<DownloadTask> list();

}

File: src/main/kotlin/cz/linkedlist/ContinuousCheckingService.kt
package cz.linkedlist

import org.springframework.scheduling.Trigger

/**
 * @author Martin Macko <https:></https:>//github.com/LinkedList>.
 */
interface ContinuousCheckingService {

    /**
     * Method for registering new continuous checking task of desired tile.
     * @param utm utm of the tile
     * @param cloudiness desired minimal cloudiness in percents
     * @param contents
     */
    fun register(utm: UTMCode, cloudiness: Double?, contents: Array<TileSet.Contents>)

    /**
     * Method for registering new continuous checking task of desired tile.
     * @param utm utm of the tile
     * @param cloudiness desired minimal cloudiness in percents
     * @param contents
     * @param cron trigger expression
     */
    fun register(utm: UTMCode, cloudiness: Double?, contents: Array<TileSet.Contents>, cron: Trigger)

    /**
     * Lists all currently registered tasks
     * @return list of all registered tasks
     */
    fun list(): Collection<DownloadTask>

}
